core-blocks/playlist/edit.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	CheckboxControl,
+	PanelBody,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	InspectorControls,
+	MediaUpload,
+	BlockControls,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			editing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	render() {
+		const { attributes, setAttributes, className } = this.props;
+		const { editing } = this.state;
+		const { tracklist, artists, images } = attributes;
+
+		const onSelectMedia = ( media ) => {
+			//check if there are returned media items and set attributes when there are
+			if ( media && media[ 0 ].url ) {
+				media = ( 1 < media.length ) ? media : [ media ];
```

Comment:
What does this line? If you select just one file, you create an array of an array?

Reply:
haha yeah I guess it does. This serves no function. I've removed it and validated that selecting one item works as we would expect. Thanks!